src/App.js

```diff
@@ -14,11 +15,108 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    baseOrder: [...goodsFromServer],
+    buttonVisibility: true,
+    listVisibility: false,
```

Comment:
```suggestion
    listVisibility: false,
```

Let's only use 'listVisibility' variable instead of two. You can render button or list depending only on one of them:

`{!listVisibility && <button>show</button>}`
and
`{listVisibility && <GoodsList />}`